lib/src/clojure_lsp/shared.clj

```diff
@@ -529,3 +531,13 @@
   expects Clojure style JSON, or when a map needs to be round-tripped from
   clojure-lsp to the client and back without case changes."
   lsp.responses/preserve-kebab-case)
+
+(defn normalize-uri-from-client [uri]
```

Comment:
I wonder if we should have this as a lsp4clj flag, it seems to me that all servers could have this encode problem from clients, so a easy flag to enable that or not may be a good idea

Reply:
Agreed, it's a shared problem. But since lsp4clj doesn't know about the structure of any of the inbound LSP messages, it'd be hard for it to do the conversion itself. This helper function would definitely be a candidate for https://github.com/clojure-lsp/lsp4clj/issues/14 though.This is another thing that might have been easier if we'd modeled the message handlers as routes instead of multimethods.